Bottom: 043fa6ab6b3ef538490004674f8581c179ee779b
Top:    437532255737821250b96fe67339f2c4badecfb9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-18 23:49:54 +0000

src/action.c: Abort the Import(setnewport,mask) action if the mask isn't set

This avoids using an uninitialised variable.

(Caught by clang)


---

diff --git a/src/action.c b/src/action.c
index 9480355..cac982a 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7700,11 +7700,11 @@ ActionImport (int argc, char **argv, Coord x, Coord y)
 	}
       else if (strcasecmp (xs, "mark") == 0)
 	{
-	  if (Marked.status)
-	    {
-	      x = Marked.X;
-	      y = Marked.Y;
-	    }
+	  if (!Marked.status)
+	    return 0;
+
+	  x = Marked.X;
+	  y = Marked.Y;
 	}
       else if (ys)
 	{
